contracts/BSCValidatorSet.sol

```diff
@@ -470,15 +557,182 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     if (m>n) {
       for (uint i = n;i<m;i++) {
         currentValidatorSet.push(validatorSet[i]);
+        validatorExtraSet.push(ValidatorExtra(0, false,
```

Comment:
it only operate the `apeend` validators, for the origin validators, we still need to reset validatorExtraSet. As you can see `currentValidatorSet[i].incoming = 0;`

Reply:
in line 575,  it is already made sure all new validators are cleared maintainInfo

```
    // make sure all new validators are cleared maintainInfo
    // should not happen, still protect
    for (uint i = 0; i < currentValidatorSet.length; i++) {
      validatorExtraSet[i].isMaintaining = false;
      validatorExtraSet[i].enterMaintenanceHeight = 0;
    }
```